src/Our.Umbraco.MeganavV8.Core/ValueConnectors/MeganavV8ValueConnector.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Deploy;
+using Umbraco.Core.Models;
+
+namespace Our.Umbraco.MeganavV8.Core.ValueConnectors
+{
+    /// <summary>
+    /// Represents a value connector for the Our.Umbraco.MeganavV8 property editor.
+    /// </summary>
+    /// <seealso cref="IValueConnector" />
+    public class MeganavV8ValueConnector : IValueConnector
+    {
+
+        public string ToArtifact(object value, PropertyType propertyType, ICollection<ArtifactDependency> dependencies)
+        {
+            var svalue = value as string;
+            if (string.IsNullOrWhiteSpace(svalue) || !svalue.DetectIsJson())
+            {
+                return null;
+            }
```

Comment:
It's probably better to return the value as-is and only parse the links/UDI of it's a valid value.

Reply:
I'm not sure on what is the best way to handle this, I haven't been able to cause the error to occur.

I've taken your advise as you seem to be more familiar with this part of the system 👍